src/modules/75-cf/components/FlagArchiving/useArchiveFlagDialog.tsx

```diff
@@ -46,36 +107,65 @@ const useArchiveFlagDialog = ({
     }
   }
 
+  const getDialogContent = () => {
+    if (state === STATUS.error)
+      return <PageError message={showError(dependentFlagsError)} onClick={() => refetchDependentFlags()} />
+
+    if (state === STATUS.loading) return <ContainerSpinner flex={{ align: 'center-center' }} />
+    if (state === STATUS.hasDependentFlags) {
+      return (
+        <CannotArchiveWarning
+          flagIdentifier={flagData.identifier}
+          flagName={flagData.name}
+          dependentFlagsResponse={dependentFlagsResponse as Features}
+          queryParams={dependentFlagQueryParams as GetDependentFeaturesQueryParams}
+          pageNumber={pageNumber}
+          setPageNumber={setPageNumber}
+        />
+      )
+    }
+    if (state === STATUS.hasNoDependentFlags) {
+      return (
+        <Layout.Vertical spacing="small">
+          <String
+            stringID="cf.featureFlags.archiving.warningDescription"
+            vars={{ flagIdentifier: flagData.identifier }}
+            useRichText
+          />
+          <Text>{getString('cf.featureFlags.archiving.confirmFlag')}</Text>
+          <TextInput
+            onPaste={e => e.preventDefault()}
+            errorText={
+              !isAnIdentifierMatch && validationFlagIdentifier !== ''
+                ? getString('cf.featureFlags.archiving.mismatchIdentifierError')
+                : undefined
+            }
+            intent={!isAnIdentifierMatch && validationFlagIdentifier !== '' ? Intent.DANGER : Intent.PRIMARY}
+            aria-label={getString('cf.featureFlags.archiving.confirmFlag')}
+            placeholder={flagData?.identifier}
+            value={validationFlagIdentifier}
+            onChange={e => {
+              setValidationFlagIdentifier((e?.target as HTMLInputElement)?.value)
+              setIsAnIdentifierMatch((e?.target as HTMLInputElement)?.value === flagData?.identifier)
+            }}
```

Comment:
If you type `e` and use `currentTarget`, you can drop the type assertions.
```suggestion
            onChange={(e:FormEvent<HTMLInputElement>) => {
              setValidationFlagIdentifier(e.currentTarget.value)
              setIsAnIdentifierMatch(e.currentTarget.value === flagData?.identifier)
            }}
```

Reply:
neat! TY